tests/test_task.py

```diff
@@ -238,26 +238,25 @@ def test_build_trials_seq_distributions(session: dict, catch_prob: float):
     task = Task(NAME, session=session, catch_prob=catch_prob)
     _ = task.generate_trials(ntrials=NTRIALS)
     assert isinstance(task._modality_seq, np.ndarray)
-    assert len(task._modality_seq) == task._ntrials
-    task._modalities.add("catch")
-    counts = {modality: np.sum(task._modality_seq == modality) for modality in task._modalities}
+
     # Assert that the counts match the expected distribution within a certain tolerance
-    assert np.isclose(counts["catch"] / len(task._modality_seq), task.catch_prob, atol=0.1)  # within 10% tolerance
-    assert np.isclose(
-        counts["v"] / len(task._modality_seq),
-        task.session["v"] - task.catch_prob * task.session["v"],
-        atol=0.1,
-    )  # within 10% tolerance
-    assert np.isclose(
-        counts["a"] / len(task._modality_seq),
-        task.session["a"] - task.catch_prob * task.session["a"],
-        atol=0.1,
-    )
-    assert np.isclose(
-        (counts["a"] + counts["v"] + counts["catch"]) / len(task._modality_seq),
-        1,
-        atol=0.1,
-    )
+    counts = {modality: np.sum(task._modality_seq == modality) for modality in [*task._modalities, "catch"]}
+    tolerance = 0.2
+    for i, comp in enumerate(
+        [
+            (counts["catch"] / len(task._modality_seq), task.catch_prob),
+            (counts["v"] / len(task._modality_seq), task.session["v"] - task.catch_prob * task.session["v"]),
+            (counts["a"] / len(task._modality_seq), task.session["a"] - task.catch_prob * task.session["a"]),
+        ],
+    ):
+        assert np.isclose(
```

Comment:
This is more compact than before, but very hard to follow. I'd revert it to three separate asserts, or use clearer variables names (e.g., for `comp`; but I don't have nice ideas about how to improve that, and I wouldn't spend time in doing so)

Reply:
I actually found it hard to follow before already. Is this better: addd4bb?
Also, the fact that we need to change the absolute tolerance to 0.2 to keep it from failing regularly makes me wonder whether this test still serves a purpose.

Reply:
Better :) I think 0.2 tolerance still serves the purpose